app/scripts/services/rowMethods.js

```diff
@@ -0,0 +1,51 @@
+'use strict';
+
+angular.module('openshiftConsole')
+  .factory('rowMethods', function() {
+
+    var expandedKey = function(apiObject) {
+      var uid = _.get(apiObject, 'metadata.uid');
+      if (!uid) {
+        return null;
+      }
+
+      return 'overview/expand/' + uid;
+    };
+
+    var setInitialExpandedState = function(row) {
+      var key = expandedKey(row.apiObject);
+      if (!key) {
+        row.expanded = false;
+        return;
+      }
+
+      var item = sessionStorage.getItem(key);
+      if (!item && row.state.expandAll) {
+        row.expanded = true;
+        return;
+      }
+
+      row.expanded = item === 'true';
+    };
+
+    return {
+      toggleExpand: function(e, always) {
+        // Don't toggle if clicking on a link inside the row unless `always` is set.
+        if (!always && $(e.target).closest("a").length > 0) {
+          return;
+        }
+
+        var key = expandedKey(this.apiObject);
+        if (!key) {
+          return;
+        }
+        console.log('we togglin');
```

Comment:
stealing one of @fabianofranz 's favorites https://www.youtube.com/watch?v=8oP6xwzV7OQ